### PR TITLE
Undergrad

### PR DESCRIPTION
**Pull request by pmarke**
Fixed several things. Add code to nodes to not publish anything to a topic until they receive data from topics they subscribe in order to avoid uninitialized variables. The parameter R_min wasn't being used so I fixed that. Lastly, the function dubins_parameters was calculating 'ell' based of the size of the vectors (cle,crs,cls,cle) instead of their norm.

### CODE REVIEW

**Comment by superjax** [comment]
Hey @pmarke, do you think you could rebase this branch on RC1.0 so we don't have conflicts to merge?

File: src/path_manager_example.cpp [hunk 1]
```diff
@@ -384,6 +419,9 @@ void path_manager_example::dubinsParameters(const waypoint_s start_node, const w
         _dubinspath.pe(2) = end_node.w[2];
         _dubinspath.chie = end_node.chi_d;
 
+        ROS_WARN_STREAM("ps: " << _dubinspath.ps);
```
Comment (gellings) [inline src/path_manager_example.cpp]: Do we need these warnings?

File: src/path_manager_example.cpp [hunk 2]
```diff
@@ -444,9 +482,17 @@ void path_manager_example::dubinsParameters(const waypoint_s start_node, const w
         if(L4 < _dubinspath.L)
         { _dubinspath.L = L4; idx = 4; }
 
+        ROS_WARN("L1: %f \n", L1);
+        ROS_WARN("L2: %f \n", L2);
+        ROS_WARN("L3: %f \n", L3);
+        ROS_WARN("L4: %f \n", L4);
+        ROS_WARN("idx: %i \n", idx);
```
Comment (gellings) [inline src/path_manager_example.cpp]: These warnings could end up being a problem

File: src/path_manager_example.cpp [hunk 3]
```diff
@@ -400,14 +438,14 @@ void path_manager_example::dubinsParameters(const waypoint_s start_node, const w
...
+        float L1 = (crs - cre).norm() + R*mo(2*M_PI_F + mo(theta - M_PI_2_F) - mo(_dubinspath.chis - M_PI_2_F))
                 + R*mo(2*M_PI_F + mo(_dubinspath.chie - M_PI_2_F) - mo(theta - M_PI_2_F));
 
         // compute L2
-        ell = (cle - crs).size();
+        ell = (cle - crs).norm();
         theta = atan2f(cle(1) - crs(1), cle(0) - crs(0));
         float L2;
-        if(2*R/ell > 1.0f || 2*R/ell < -1.0f)
+        if(2*R > ell)
```
Comment (gellings) [inline src/path_manager_example.cpp:438]: Has this been tested?

Comment (pmarke) [reply src/path_manager_example.cpp:438]: It has been tested. I don't know why the other statement is necessary since R and ell are always positive.

**Pull request closed without merging**